### PR TITLE
Bugfix/group offer volunteers edit form as department manager not working

### PR DESCRIPTION
**Pull request by casaper**
# [Story in Trello](https://trello.com/c/QEk7bZuv/21-bug-group-offer-edit-existing-nested-attribute-volunteers-fields-are-displayed-but-not-filled-with-data-so-on-submit-the-form-is)

### Done for approval?

* [x] Acceptance criteria are met
* [ ] Code quality checks are green
* [ ] Readme updated if needed
* [x] Story under test
* [ ] Mobile works
* [ ] Seeds created
* [ ] Translated
* [ ] Code is reviewed

### Done after the merge?

* [ ] Deployed to staging after the merge
* [ ] Tested manually on staging

### CODE REVIEW

File: db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class GroupAssignmentUniquenessGroupOfferCombinedWithVolunteer < ActiveRecord::Migration[5.1]
+  def change
+    add_column :group_assignments, :active, :boolean, default: true
```
Comment (nuritnt) [inline db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb]: for what is this needed in the group_assignments? isn't it already handled in the group offer itself?

Comment (casaper) [reply db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb]: That is the recommended way, to do it as well in the db.

File: app/models/group_assignment.rb [hunk 2]
```diff
@@ -2,12 +2,16 @@ class GroupAssignment < ApplicationRecord
   include GroupAssignmentAndAssignmentScopes
 
   belongs_to :group_offer
+  validates_presence_of :group_offer
```
Comment (chrysalag) [inline app/models/group_assignment.rb]: Why is this [needed](https://blog.bigbinary.com/2016/02/15/rails-5-makes-belong-to-association-required-by-default.html)?

File: db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb (hunk 1 above)
Comment (chrysalag) [reply db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb]: What is meant by active?

**Comment by casaper** [comment]
@chrysalag 
> What is meant by active?

This is working with paranoia gem, same as with the Users table. This way only not softdeleted records are checked for uniqueness. deleted records are not throwing an error if they'd duplicate with a not soft deleted one

File: db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb (hunk 1 above)
Comment (casaper) [reply db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb]: This is working with paranoia gem, same as with the Users table. This way only not softdeleted records are checked for uniqueness. deleted records are not throwing an error if they'd duplicate with a not soft deleted one

Comment (casaper) [reply db/migrate/20171026145935_group_assignment_uniqueness_group_offer_combined_with_volunteer.rb]: the field name is given by paranoiagem

File: app/models/group_assignment.rb (hunk 2 above)
Comment (casaper) [reply app/models/group_assignment.rb]: its not, its rubbel

**Pull request merged**